detection_rules/rule.py

```diff
@@ -30,14 +35,32 @@ class RuleMeta(MarshmallowDataclassMixin):
     deprecation_date: Optional[definitions.Date]
 
     # Optional fields
-    beats_version: Optional[definitions.BranchVer]
-    ecs_versions: Optional[List[definitions.BranchVer]]
     comments: Optional[str]
     maturity: Optional[definitions.Maturity]
+    min_stack_version: Optional[definitions.SemVer]
     os_type_list: Optional[List[definitions.OSType]]
     query_schema_validation: Optional[bool]
+    query_validate_master: Optional[bool]
     related_endpoint_rules: Optional[List[str]]
 
+    def get_validation_stack_versions(self) -> Dict[str, dict]:
+        """Get a dict of beats and ecs versions per stack release."""
+        stack_versions = {}
+
+        if self.min_stack_version:
+            stack_versions = get_stack_schemas(self.min_stack_version)
+
+        # if min_stack_version is not defined in a rule or the stack version is not defined in the stack-schema-map,
+        #  then validate with the latest non-master schemas saved locally
```

Comment:
I don't think this is quite right. If `min_stack_version` is missing, then we know it was some time prior to 7.13.
So we want to validate everything from 7.13 to <current_stack_version>.

Basically, if that field is missing then we set it to `min(Version(current_version), Version("7.13"))`, in memory. We don't want to modify the dataclass though, because then `toml-lint` will save it back.

Reply:
Fixed in 7a90cc3